packages/core/src/lib/message/version_0.ts

```diff
@@ -37,6 +43,20 @@ export class DecodedMessage implements IDecodedMessage {
     return this.proto.contentTopic;
   }
 
+  public get hash(): Uint8Array {
+    if (this._hash === undefined) {
+      this._hash = messageHash(this.pubsubTopic, this.proto as IProtoMessage);
+    }
+    return this._hash;
+  }
+
+  public get hashStr(): string {
+    if (this._hashStr === undefined) {
+      this._hashStr = bytesToHex(this.hash);
```

Comment:
```suggestion
      this._hashStr = bytesToHex(this._hash);
```

also, what happens if `hashStr()` is called before `this._hash` is set?